nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessGroupResource.java

```diff
@@ -3301,6 +3307,33 @@ public Response instantiateTemplate(
         );
     }
 
+    private void verifyPublicPortUniqueness(FlowSnippetDTO snippet) {
+        snippet.getInputPorts().stream().filter(portDTO -> Boolean.TRUE.equals(portDTO.isAllowRemoteAccess()))
```

Comment:
`filter(portDTO -> Boolean.TRUE.equals(portDTO.isAllowRemoteAccess()))` could be simplified to

`filter(PortDTO::isAllowRemoteAccess)`

Reply:
Yeah, but it produces NullPointerException when the value is null. The intent of using non primitive type was accepting client requests without having this field. DTO usually doesn't have value if the REST client didn't intended to update or set.